include/util/string_view.hpp

```diff
@@ -0,0 +1,34 @@
+#ifndef OSRM_STRING_VIEW_HPP
+#define OSRM_STRING_VIEW_HPP
+
+#include <boost/functional/hash.hpp>
+#include <boost/utility/string_ref.hpp>
+
+namespace osrm
+{
+namespace util
+{
+// Convenience typedef: boost::string_ref, boost::string_view or C++17's string_view
+using StringView = boost::string_ref;
+
+} // namespace util
+} // namespace osrm
+
+// Specializing hash<> for user-defined type in std namespace, this is standard conforming.
+namespace std
+{
+template <> struct hash<::osrm::util::StringView> final
+{
+    std::size_t operator()(::osrm::util::StringView v) const noexcept
```

Comment:
`StringView` is basically `struct StringView { const char* first; const char* last; }` - better to pass by value. The `StringView` is immutable by default, too, so there is no real benefit of passing it by `const`.

(If you're asking yourself, why immutable by default, consider `StringView s{"hello"}; s[0] = 'w'; // oops`)